### PR TITLE
Admin: add improved product copy

### PR DESCRIPTION
**Pull request by tulimaki**
Copy supplier, categories, media, stock managed value and customer group price

### CODE REVIEW

File: shuup/admin/modules/products/views/copy.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                new_product.primary_image = media
+                new_product.save()
+
+        for price in CgpPrice.objects.filter(product=product, shop=shop_product.shop):
+            CgpPrice.objects.create(
+                product=new_product, shop=shop_product.shop, group=price.group,
+                price_value=price.price_value
+            )
+
+        for supplier in shop_product.suppliers.all():
```
Comment (chessbr) [inline shuup/admin/modules/products/views/copy.py:68]: Why the stocks should be copied to this new product? I don't think it makes sense. Also, don't import simple_supplier here as this module can be replaced by any other. If you need to add stocks, use the interface of the supplier module: `get_stock_status` and `adjust_stock`.

File: shuup/admin/modules/products/views/copy.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        new_product.save()
+
+        new_shop_product = ShopProduct.objects.create(product=new_product, shop=shop_product.shop)
+        new_shop_product.visibility = shop_product.visibility
+        new_shop_product.purchasable = shop_product.purchasable
+        new_shop_product.default_price_value = shop_product.default_price_value
+        new_shop_product.primary_category = shop_product.primary_category
+        new_shop_product.save()
+
+        new_shop_product.suppliers = shop_product.suppliers.all()
```
Comment (chessbr) [inline shuup/admin/modules/products/views/copy.py:49]: use `.set()` instead

File: shuup/admin/modules/products/views/copy.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        new_shop_product.categories = shop_product.categories.all()
```
Comment (chessbr) [inline shuup/admin/modules/products/views/copy.py:50]: use `.set()` instead

File: shuup/admin/modules/products/views/copy.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                price_value=price.price_value
+            )
+
+        for supplier in shop_product.suppliers.all():
+            origin_product_stock_count = StockCount.objects.get_or_create(supplier=supplier, product=product)[0]
+            new_product_stock_count = StockCount.objects.get_or_create(supplier=supplier, product=new_product)[0]
+            new_product_stock_count.stock_managed = origin_product_stock_count.stock_managed
+            new_product_stock_count.save(update_fields=["stock_managed"])
+
+        messages.success(request, _("Product %s copy successfull.") % new_product)
```
Comment (chessbr) [inline shuup/admin/modules/products/views/copy.py:74]: I would recommend you to trigger some custom signal here like:
```py
product_copied.send(sender=type(self), copied=product, copy=new_prouct)
```

this way, other addons can do their copy part if they need to (this is the case of the supplier module and others).

File: shuup/admin/modules/products/views/copy.py (hunk 1 above)
Comment (tulimaki) [reply shuup/admin/modules/products/views/copy.py:68]: It does not copy the stock but the stock managed value. But yes maybe this should be only done when the simple supplier is actually installed.

**Pull request merged**